LiteCore/Storage/SQLiteDataFile.hh

```diff
@@ -174,9 +174,10 @@ namespace litecore {
 
             WithNewDocs = 400,  // New document/revision storage (CBL 3.0)
 
-            WithDeletedTable   = 500,  // Added 'deleted' KeyStore for deleted docs (CBL 3.0?)
-            WithIndexesLastSeq = 501,  // Added 'lastSeq' column to 'indexes' table (CBL 3.2)
-            MaxReadable        = 599,  // Cannot open versions newer than this
+            WithExpirationColumn = 490,  // Added 'expiration' column to KeyStore
```

Comment:
Can you insert a new version before existing versions? How would a version at 500 or 501 get upgraded? There should be a test, I think.

Reply:
Any database at version >=500 already has the expiration column, because `addExpiration` was added to `KeyStore::reopen` in the same commit that `WithDeletedTable (500)` migration was added.

Reply:
So you actually try to retrofit into current upgrade path. This relies on very precise understanding of other part of code and the schema. 
Recognizing this fact, we can also treat it as a new version, but because of the above fact, do
ALTER TABLE ..... IF NOT EXISTS 
I am little concerned of retrofitting a new version into the history

Reply:
> ALTER TABLE ..... IF NOT EXISTS
There is no such syntax. We need to check the existence of the column manually

Reply:
If you check the existence, do we still need to insert a new version into past history? This is the new version, after this new version, we will know schema has the expiration table at the time of creation.

Reply:
What's the code path that supports your statement? Is it in the same transaction of the upgrade transaction? Is it possible that some failure occurs outside the upgrade transaction that arrives following state:
- version bumped 500
- some keystore not opened successfully and expiration not added successfully.
Since you removed the addExpiration completely, there is no way to have the column.

Reply:
SQLiteKeyStore::reopen calls `addExpiration` (on the main branch).
So if a database is version 500, every collection which has been used since it upgraded to version 500 has the expiration column.

Reply:
"which has been used since ..." -- what if not used?

Reply:
I don't think it's worth trying to optimize. Seems like you need to make WithExpiration = 502 and always do it (once) for every database, just to make sure.